hive-metastore/src/test/java/org/apache/iceberg/hive/HiveTableTest.java

```diff
@@ -386,6 +386,12 @@ public void testHiveTableAndIcebergTableWithSameName(TableType tableType)
 
     assertThat(catalog.tableExists(TABLE_IDENTIFIER)).isTrue();
     HIVE_METASTORE_EXTENSION.metastoreClient().dropTable(DB_NAME, hiveTableName);
+
```

Comment:
We usually try to create a different test for every feature. In this case, if a test fails, it is easier to understand what went wrong.

Reply:
Yeah I I agree with you on different test for new feature.  But in this particular case, it was already part of dropTable tests in https://github.com/apache/iceberg/blob/a52afdc484f9e63477b96d0182586692daac8eab/core/src/test/java/org/apache/iceberg/catalog/CatalogTests.java#L838-L845 and `catalog.tableExists` was tested pretty extensively in all other existing unit tests as well.

Reply:
We are complicate the implementation of the feature. For me this suggests that we would be better off with a separate trst

Reply:
Unrelated, but this test also verifies that there is an existing table remaining from the previous test. All these tests would be much simpler if they cleaned up after themselves.